### PR TITLE
470-default behaviour of arraydataset when the input is single sequence

### LINKED ISSUE #470: default behaviour of arraydataset when the input is single sequence
**Is your feature request related to a problem? Please describe.**
when `ArrayDataset` is from a single sequence, the outputs shouldn't be in nested lists.
for example:
```python
from monai.data import ArrayDataset
from torch.utils.data import DataLoader

y_train = "abcd"
ds = ArrayDataset(y_train)
for x in DataLoader(ds, batch_size=2):
    print(x)
```
currently outputs:
```
[('a', 'b')]
[('c', 'd')]
```
but I'd like to have:
```
['a', 'b']
['c', 'd']
```
**Describe alternatives you've considered**
use `monai.data.Dataset` directly:
```python
y_train = "abcd"
ds = Dataset(y_train)
for x in DataLoader(ds, batch_size=2):
    print(x)
```
but it doesn't support randomised transforms

### PR DESCRIPTION
**Pull request by wyli**
Fixes #470.

### Description
when the input is a single sequence (such as `img`, instead of multiple sequences such as `img`, `seg`, `labels`), ArrayDataset becomes a randomizable version of `monai.data.Dataset`

### Status
**Ready**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New tests added to cover the changes
- [x] Docstrings/Documentation updated

### CODE REVIEW

File: monai/data/dataset.py [hunk 1]
```diff
@@ -374,41 +372,58 @@ def __call__(self, input_):
...
-        for dataset in self.datasets:
-            if isinstance(dataset.transform, Randomizable):
-                dataset.transform.set_random_state(seed=self.seed)
-        return super().__getitem__(index)
+        if isinstance(self.dataset, ZipDataset):
+            for dataset in self.dataset.datasets:
+                if isinstance(getattr(dataset, "transform", None), Randomizable):
+                    dataset.transform.set_random_state(seed=self._seed)
+        if isinstance(getattr(self.dataset, "transform", None), Randomizable):
+            self.dataset.transform.set_random_state(seed=self._seed)  # noqa: T484
```
Comment (Nic-Ma) [inline monai/data/dataset.py:428]: What's this T484 here? no QA for T484?

**Comment by wyli** [comment]
@Nic-Ma thanks for the review, I made a few changes:
- apply_transform function (so that it supports transform(list), useful for zipdataset):
https://github.com/Project-MONAI/MONAI/blob/7e8ebb361d5c46ee2f7cf674282b9024c12a9f33/monai/transforms/utils.py#L240
- added a unit test for NiftiDataset
- zipdataset inherits dataset (remove the #noqa line, also I think zipdatset is itself a dataset):
https://github.com/Project-MONAI/MONAI/blob/7e8ebb361d5c46ee2f7cf674282b9024c12a9f33/monai/data/dataset.py#L295
